### ISSUE TITLE
Reactive Cluster `MGET` is not running in parallel

### ISSUE THREAD

**Issue by koisyu** [redis/lettuce#2395]
## Bug Report

RedisAdvancedClusterReactiveCommandsImpl.mget is not running in parallel

#### Current Behavior

If you run the code below, it will take a long time.
```java
RedisClusterClient client = RedisClusterClient.create("redis://localhost");
RedisAdvancedClusterReactiveCommands<String, String> reactiveCommands = client.connect().reactive();
List<String> keys = IntStream.rangeClosed(1, 100).boxed().map(index -> "key-" + index).collect(Collectors.toList());
reactiveCommands.mget(keys.toArray(new String[]{})).collectList().block();
```
If I run the above code and check the Tcpdump, I can see that the requests and responses are alternating in order, like this.
That is, they are executed sequentially.

<img width="1243" alt="image" src="https://user-images.githubusercontent.com/17047078/233683439-5c199092-ad6a-4c19-9696-1e379425f391.png">

#### Expected behavior/code

I expect MGET commands to be executed in parallel.

<img width="1240" alt="image" src="https://user-images.githubusercontent.com/17047078/233685829-c25c1a94-42aa-4503-9358-3f5f61b43dc2.png">

#### Environment

- Lettuce version(s): 6.1.9.RELEASE
- Redis version: 4.0.14

#### Possible Solution

If the [Flux#concat](https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#concat-org.reactivestreams.Publisher%2E%2E%2E-) code is replaced with [Flux#mergeSequential](https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#mergeSequential-int-org.reactivestreams.Publisher%2E%2E%2E-), the MGET commands are executed in parallel.

* Flux#concat
![image](https://user-images.githubusercontent.com/17047078/233689655-ab140d1b-0edb-468d-bd5e-872fd6ff64ce.png)
* Flux#mergeSequential
![image](https://user-images.githubusercontent.com/17047078/233689722-191f7f02-58e0-4d37-be47-094f2d0270ec.png)

**Comment by koisyu** [comment]
I was worried that my explanation wasn't clear enough, so I added an example of the difference in repeated runs.
For example, if you run the code below, you can clearly see the difference.

```java
RedisClusterClient client = RedisClusterClient.create("redis://localhost");
StatefulRedisClusterConnection<String, String> connect = client.connect();
RedisAdvancedClusterReactiveCommands<String, String> reactiveCommands = connect.reactive();
String[] keys = IntStream.rangeClosed(1, 500).boxed().map(index -> "key-" + index).toArray(String[]::new);
reactiveCommands.mget(keys).collectList().block();

long startTimeInMillis = System.currentTimeMillis();
for (int i = 0; i < 100; i++) {
    reactiveCommands.mget(keys).collectList().block();
}
long endTimeInMillis = System.currentTimeMillis();
System.out.println(String.format("Total time in millis : %s", (endTimeInMillis - startTimeInMillis)));
```

Here are the execution results of the code in both cases. The execution times vary slightly depending on the conditions, but the difference between the two is always significant.

* **Flux#concat**
  * `Total time in millis : 8289`
<img width="662" alt="image" src="https://user-images.githubusercontent.com/17047078/233733004-8f8c02dd-e0f3-4ee2-9eea-7496d00c30d9.png">

* **Flux#mergeSequential**
  * `Total time in millis : 563`
<img width="689" alt="image" src="https://user-images.githubusercontent.com/17047078/233733624-4d02e907-c951-4a11-b304-060fb4e3e79b.png">

**Comment by wenhaozhao** [comment]
@mp911de Can fix the same bug to version 6.1.x ?

**Comment by tishun** [comment]
> @mp911de Can fix the same bug to version 6.1.x ?

Hey @wenhaozhao , is this still relevant or can we close this issue?

**Comment by wenhaozhao** [comment]
> > @mp911de Can fix the same bug to version 6.1.x ?
> 
> Hey @wenhaozhao , is this still relevant or can we close this issue?

sure, please

**Comment by tishun** [comment]
> sure, please

Lettuce 6.1.11.RELEASE should have the fix available
